library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -461,21 +522,31 @@ private boolean purchase(Activity activity, List<String> oldProductIds, String p
 			Bundle bundle;
 			if (oldProductIds != null && purchaseType.equals(Constants.PRODUCT_TYPE_SUBSCRIPTION))
 			{
-				bundle =
-						billingService.getBuyIntentToReplaceSkus(Constants.GOOGLE_API_SUBSCRIPTION_CHANGE_VERSION,
-																 contextPackageName,
-																 oldProductIds,
-																 productId,
-																 purchaseType,
-																 purchasePayload);
+				int apiVersion = Constants.GOOGLE_API_EXTRA_PARAMS_SUPPORTED_VERSION;
+
+				if (extraParamsBundle == null)
+				{
+					extraParamsBundle = new Bundle();
+				}
+
+				if (!extraParamsBundle.containsKey(Constants.EXTRA_PARAMS_KEY_SKU_TO_REPLACE))
+				{
+					extraParamsBundle.putStringArrayList(Constants.EXTRA_PARAMS_KEY_SKU_TO_REPLACE,
+							new ArrayList<String>(oldProductIds));
+				}
+
+				if(extraParamsBundle.containsKey(Constants.EXTRA_PARAMS_KEY_VR))
+				{
+					apiVersion = Constants.GOOGLE_API_VR_SUPPORTED_VERSION;
+				}
+
+				bundle = billingService.getBuyIntentExtraParams(apiVersion, contextPackageName, productId,
+						purchaseType, purchasePayload, extraParamsBundle);
 			}
 			else
 			{
-				bundle = billingService.getBuyIntent(Constants.GOOGLE_API_VERSION,
-													 contextPackageName,
-													 productId,
-													 purchaseType,
-													 purchasePayload);
+				bundle = billingService.getBuyIntentExtraParams(Constants.GOOGLE_API_EXTRA_PARAMS_SUPPORTED_VERSION,
```

Comment:
same here. if we change API version from v3 to v6 we must ensure we have a  backward compatibility plan